tests/units/plugins/filter/test_merge.py

```diff
@@ -0,0 +1,258 @@
+                    "templates": ["bar/dc.yml", "bar/svc.yml"],
+                },
+                {
+                    "name": "baz",
+                    "configs": ["baz.conf"],
+                    "templates": ["baz/dc.yml", "baz/svc.yml", "baz/ep.yml"],
+                },
+            ],
+            "volumes": ["volumes/foo_bar.yml", "volumes/foo_baz.yml"],
+        }
+
+        new = {"name": "foo", "services": []}
+
+        self.assertDictDeepEqual(merge_with_app(base, new), base)
+
+    def test_services_merge_without_duplicates(self):
+        """
+            Test app services merge does not creates duplicates if same items
+            appear in base and new apps
+        """
+
+        base = {
+            "name": "foo",
+            "services": [
+                {
+                    "name": "bar",
+                    "configs": ["bar.conf"],
+                    "templates": ["bar/dc.yml", "bar/svc.yml"],
+                },
+                {
+                    "name": "baz",
+                    "configs": ["baz.conf"],
+                    "templates": ["baz/dc.yml", "baz/svc.yml", "baz/ep.yml"],
+                },
+            ],
+            "volumes": ["volumes/foo_bar.yml", "volumes/foo_baz.yml"],
+        }
+
+        self.assertDictDeepEqual(merge_with_app(base, base), base)
+
+    def test_services_merge(self):
+        """Test app services merge"""
+
+        base = {
+            "name": "foo",
+            "services": [
+                {
+                    "name": "bar",
+                    "configs": ["bar.conf"],
+                    "templates": ["bar/dc.yml", "bar/svc.yml"],
+                },
+                {
+                    "name": "baz",
+                    "configs": ["baz.conf"],
+                    "templates": ["baz/dc.yml", "baz/svc.yml", "baz/ep.yml"],
+                },
+            ],
+            "volumes": ["volumes/foo_bar.yml", "volumes/foo_baz.yml"],
+        }
+
+        new = {
+            "name": "foo",
+            "services": [
+                {
+                    "name": "bar",
+                    "configs": ["bar2.conf"],
+                    "templates": ["bar/dc.yml", "bar/ep.yml"],
+                },
+                {
+                    "name": "fun",
+                    "configs": ["fun.conf"],
+                    "templates": ["fun/dc.yml", "fun/svc.yml"],
+                },
+            ],
+            "volumes": ["volumes/foo_fun.yml"],
+        }
+
+        expected = {
+            "name": "foo",
+            "services": [
+                {
+                    "name": "bar",
+                    "configs": ["bar.conf", "bar2.conf"],
+                    "templates": ["bar/dc.yml", "bar/svc.yml", "bar/ep.yml"],
+                },
+                {
+                    "name": "baz",
+                    "configs": ["baz.conf"],
+                    "templates": ["baz/dc.yml", "baz/svc.yml", "baz/ep.yml"],
+                },
```

Comment:
So the fun service is gone is it what you expect?

Reply:
Yes, we take data from right (new) to left (base) if the app is defined in left (base), _i.e._ you cannot define a new app that is not listed as available (with files in apps/).

Reply:
For the moment I just read the tests. Maybe it deserves a line in the docstring. 
Then the concept could be explained in some documentation (in another PR is fine)?